api/discovery/discovery.proto

```diff
@@ -55,7 +55,10 @@ service Discovery {
   }
 }
 
-message StartDiscoveryRequest {}
+message StartDiscoveryRequest {
+  // A list of CSPs to be discovered
+  repeated string csp = 1;
```

Comment:
I would suggest `providers` as a property name.

Reply:
Perfect, that I was searching for...